types/snuownd/index.d.ts

```diff
@@ -0,0 +1,496 @@
+// Project: http//github.com/gamefreak/snuownd
+// Definitions by: eritbh <https://github.com/eritbh>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace SnuOwnd;
+
+// These definitions use some utility classes which shouldn't be exported
+export {};
+
+// Constructs a type from `T` where all properties additionally accept `null`
+type NullableValues<T> = {[K in keyof T]: T[K] | null};
+
+/** A string buffer wrapper because JavaScript doesn't have mutable strings. */
+export class Buffer {
+    s: string;
+    /** @param str Optional string to initialize the Buffer with. */
+    constructor(str?: string);
+    truncate(size: number): void;
+}
+
+// #region Flags
+
+// #region HTML processing flags
+export const HTML_SKIP_HTML = 1;
+export const HTML_SKIP_STYLE = 2;
+export const HTML_SKIP_IMAGES = 4;
+export const HTML_SKIP_LINKS = 8;
+export const HTML_EXPAND_TABS = 16;
+export const HTML_SAFELINK = 32;
+export const HTML_TOC = 64;
+export const HTML_HARD_WRAP = 128;
+export const HTML_USE_XHTML = 256;
+export const HTML_ESCAPE = 512;
+export const HTML_ALLOW_ELEMENT_WHITELIST = 1024;
+// #endregion HTML processing flags
+
+// #region Markdown processing flags
+export const MKDEXT_NO_INTRA_EMPHASIS = 1;
+export const MKDEXT_TABLES = 2;
+export const MKDEXT_FENCED_CODE = 4;
+export const MKDEXT_AUTOLINK = 8;
+export const MKDEXT_STRIKETHROUGH = 16;
+// export const MKDEXT_LAX_HTML_BLOCKS = 32; // removed
+export const MKDEXT_SPACE_HEADERS = 64;
+export const MKDEXT_SUPERSCRIPT = 128;
+export const MKDEXT_LAX_SPACING = 256;
+export const MKDEXT_NO_EMAIL_AUTOLINK = 512;
+// #endregion Markdown processing flags
+
+// Snudown processing flags
+export const SD_AUTOLINK_SHORT_DOMAINS = 1;
+
+// Default flag sets
+export const DEFAULT_BODY_FLAGS = 805; // HTML_SKIP_HTML | HTML_SKIP_IMAGES | HTML_SAFELINK | HTML_ESCAPE | HTML_USE_XHTML;
+export const DEFAULT_WIKI_FLAGS = 1825; // HTML_SKIP_HTML | HTML_SAFELINK | HTML_ALLOW_ELEMENT_WHITELIST | HTML_ESCAPE | HTML_USE_XHTML;
+
+// #endregion Flags
+
+// Default element and attribute allowlists
+export const DEFAULT_HTML_ELEMENT_WHITELIST: string[];
+export const DEFAULT_HTML_ATTR_WHITELIST: string[];
+
+// #region Enums
+
+// Constants used by the autolink callback for identifying the type of link
+/** used internally when it is not an autolink */
+export const MKDA_NOT_AUTOLINK = 0;
+/** normal http/http/ftp/mailto/etc link */
+export const MKDA_NORMAL = 1;
+/** e-mail link without explit mailto: */
+export const MKDA_EMAIL = 2;
+
+/** The type of an autolink being processed by the renderer */
+export type MKDAAutolinkType = 1 | 2;
+
+// #endregion
+
+// #region Render callbacks
+
+// Non-exported interface listing all possible callback properties. In practice
+// there is never any requirement that all these callbacks are present, so this
+// is passed through `NullableValues<T>` and/or `Partial<T>` before being used.
+interface AllCallbacks {
+	/**
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/snuownd/index.d.ts;
  Line: 85;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

Reply:
[33c7a61](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66335/commits/33c7a617ad0a07b3c683b97b10a285ad02c2264c)